src/components/main/BackgroundImage.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { APP_SETTINGS } from '../../constants/constants';
+import { useAppSettingFile, useAppSettings } from '../context/appData';
+
+const useStyles = makeStyles(() => ({
+  image: {
+    width: '100%',
+    height: '100%',
+    display: 'block',
+    backgroundSize: '100% 100%'
+  },
+}));
+
+const BackgroundImage = () => {
+  const classes = useStyles();
+  const { data: appSettings } = useAppSettings();
+  const backgroundSetting = appSettings?.find(
+    ({ name }) => name === APP_SETTINGS.BACKGROUND,
+  );
+  const { data: backgroundImage } = useAppSettingFile(
+    backgroundSetting?.id,
+    Boolean(backgroundSetting),
+  );
+
+  if (!backgroundSetting || !backgroundImage) {
+    return null;
+  }
+
+  const url = URL.createObjectURL(backgroundImage);
+
+  if (backgroundSetting.data?.extra?.file) {
```

Comment:
Now I see it, I think there's a small issue. `useAppSettingFile` suppose by default the setting you provide is a file. So there's no real need to check afterwards if the setting is a file (especially since you don't do anything if it isn't a file). I would instead play with the `enabled` property (here set as `Boolean(x)`) and change it for `Boolean(backgroundSetting?.data?.extra?.file)`. That way it will only fetch the image if `backgroundSetting` exists and is a file. I guess then you don't need line 32, nor 43-45.